### PR TITLE
Remove floating point types from radix sort fast-path

### LINKED ISSUE #7212: [BUG] NaN values no longer sort correctly after optimization
**Describe the bug**
After #7167 was merged in NaN values no longer sort the same as they did before.  From https://github.com/rapidsai/cudf/pull/7167#issuecomment-767687989 it sounds like the optimization will be disabled for floating point sorts.

### PR DESCRIPTION
**Pull request by davidwendt**
Closes #7212 

Reference https://github.com/rapidsai/cudf/pull/7167#issuecomment-767687989

Using radix sort for all fixed-width types causes an [error in Spark when floating point columns contain NaN elements](https://github.com/NVIDIA/spark-rapids/issues/1585).

This PR removes floating-point column types from the radix fast-path. This means the original `relational_compare` row operator is used to handle sorting floating point columns since they could possibly contain NaN elements.

The `NANSorting` gtest included null elements so it did not catch the fast-path output discrepancy. This PR adds a `NANSortingNonNull` gtest to check for the desired NaN sorting behavior.

### CODE REVIEW

**Review by jrhemstad — APPROVED** [review]

File: cpp/src/sort/sort_column.cu [hunk 1]
```diff
@@ -32,15 +43,15 @@ struct column_sorted_order_fn {
...
                   mutable_column_view& indices,
                   bool ascending,
                   rmm::cuda_stream_view stream)
   {
-    // A non-stable sort on a fixed-width column with no nulls will use a radix sort
-    // if using only the thrust::less or thrust::greater comparators but also
-    // requires making a copy of the input data.
+    // A non-stable sort on a column of arithmetic type with no nulls will use a radix sort
+    // if specifying only the `thrust::less` or `thrust::greater` comparators.
+    // But this also requires making a copy of the input data.
```
Comment (codereport) [inline cpp/src/sort/sort_column.cu:54]: This reads a bit odd.

**Review by codereport — APPROVED** [review]
Looks good. Might suggest rewording the comment.

**Comment by revans2** [comment]
I reran the tests that we saw failing and this has fixed them.  Thanks.

**Comment by davidwendt** [comment]
@gpucibot merge
